library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -0,0 +1,652 @@
+  {
+    this.queueCapacity = queueCapacity;
+  }
+
+  /**
+   * Returns the ordered key used to generate the range queries
+   */
+  public String getKey()
+  {
+    return key;
+  }
+
+  /**
+   * Sets the ordered key used to generate the range queries
+   */
+  public void setKey(String key)
+  {
+    this.key = key;
+  }
+
+  /**
+   * Returns the tableName which would be queried
+   */
+  public String getTableName()
+  {
+    return tableName;
+  }
+
+  /**
+   * Sets the tableName to query
+   */
+  public void setTableName(String tableName)
+  {
+    this.tableName = tableName;
+  }
+
+  /**
+   * Returns rangeQueryPair - <lowerBound,upperBound>
+   */
+  public KeyValPair<String, String> getRangeQueryPair()
+  {
+    return rangeQueryPair;
+  }
+
+  /**
+   * Sets the rangeQueryPair <lowerBound,upperBound>
+   */
+  public void setRangeQueryPair(KeyValPair<String, String> rangeQueryPair)
+  {
+    this.rangeQueryPair = rangeQueryPair;
+  }
+
+  /**
+   * Returns batchSize indicating the number of elements in emitQueue
+   */
+  public int getBatchSize()
+  {
+    return batchSize;
+  }
+
+  /**
+   * Sets batchSize for number of elements in the emitQueue
+   */
+  public void setBatchSize(int batchSize)
+  {
+    this.batchSize = batchSize;
+  }
+
+  public AbstractJdbcPollInputOperator()
+  {
+    this.pollInterval = 10 * 1000;
+    this.fetchSize = 10000;
+    currentWindowRecoveryState = new MutablePair<>();
+    windowManager = new FSWindowDataManager();
+  }
+
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    spinMillis = context.getValue(Context.OperatorContext.SPIN_MILLIS);
+    execute = true;
+    cause = new AtomicReference<Throwable>();
+    emitQueue = new ArrayBlockingQueue<List<T>>(queueCapacity);
+    this.context = context;
+    operatorId = context.getId();
+
+    try {
+
+      //If its a range query pass upper and lower bounds
+      //If its a polling query pass only the lower bound
+      if (getRangeQueryPair().getValue() != null) {
+        ps = store.getConnection()
+            .prepareStatement(
+                JdbcMetaDataUtility.buildRangeQuery(getTableName(), getKey(), rangeQueryPair.getKey(),
+                    rangeQueryPair.getValue()),
+                java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+      } else {
+        ps = store.getConnection().prepareStatement(
+            JdbcMetaDataUtility.buildPollableQuery(getTableName(), getKey(), rangeQueryPair.getKey()),
+            java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+        isPollable = true;
+      }
+
+    } catch (SQLException e) {
+      LOG.error("Exception in initializing the range query for a given partition", e);
+      throw new RuntimeException();
+    }
+
+    windowManager.setup(context);
+    LOG.debug("super setup done...");
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
```

Comment:
Good point.will add.

Thanks,
Dev
On May 26, 2016 2:43 PM, "gauravgopi123" <notifications@github.com> wrote:

> In
> library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java
> <https://github.com/apache/incubator-apex-malhar/pull/282#discussion_r64826243>
> :
>
> > +            java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
> > +        isPollable = true;
> > +      }
> > +
> > +    } catch (SQLException e) {
> > +      LOG.error("Exception in initializing the range query for a given partition", e);
> > +      throw new RuntimeException();
> > +    }
> > +
> > +    windowManager.setup(context);
> > +    LOG.debug("super setup done...");
> > +  }
> > +
> > +  @Override
> > +  public void beginWindow(long windowId)
> > +  {
>
> begin window will not be called when checkpoint happens within
> Applicationwindow so how would replay happen?
>
> May be you want to set @OperatorAnnotation(checkpointableWithinAppWindow =
> false) on the operator
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-apex-malhar/pull/282/files/a96c06d6f4a3d1c5f4a69340ca1c9dce3d45a7e1#r64826243>
>